packages/core/src/StarknetWindowObject.ts

```diff
@@ -0,0 +1,123 @@
+import type { AccountInterface, ProviderInterface } from "starknet"
+
+export type AccountChangeEventHandler = (accounts: string[]) => void
+
+export type NetworkChangeEventHandler = (network?: string) => void
+
+export type WalletEventHandlers =
+  | AccountChangeEventHandler
+  | NetworkChangeEventHandler
+
+export type WalletEvents =
+  | {
+      type: "accountsChanged"
+      handler: AccountChangeEventHandler
+    }
+  | {
+      type: "networkChanged"
+      handler: NetworkChangeEventHandler
+    }
+
+// EIP-747:
+// https://github.com/ethereum/EIPs/blob/master/EIPS/eip-747.md
+export interface WatchAssetParameters {
+  type: "ERC20" // The asset's interface, e.g. 'ERC20'
+  options: {
+    address: string // The hexadecimal StarkNet address of the token contract
+    symbol?: string // A ticker symbol or shorthand, up to 5 alphanumerical characters
+    decimals?: number // The number of asset decimals
+    image?: string // A string url of the token logo
+    name?: string // The name of the token - not in spec
+  }
+}
+
+// EIP-3085
+// https://github.com/ethereum/EIPs/blob/master/EIPS/eip-3085.md
+
+export interface AddStarknetChainParameters {
+  id: string
+  chainId: string // A 0x-prefixed hexadecimal string
+  chainName: string
+  baseUrl: string
+  rpcUrl?: string
+  blockExplorerUrl?: string
+  accountImplementation?: string
+
+  nativeCurrency?: {
+    name: string
+    symbol: string // 2-6 characters long
+    decimals: 18
+  } // Currently ignored.
+  iconUrls?: string[] // Currently ignored.
+}
+
+export interface SwitchStarknetChainParameter {
+  chainId: string // A 0x-prefixed hexadecimal string
+}
+
+export type RpcMessage =
+  | {
+      type: "wallet_watchAsset"
+      params: WatchAssetParameters
+      result: boolean
+    }
+  | {
+      type: "wallet_addStarknetChain"
+      params: AddStarknetChainParameters
+      result: boolean
+    }
+  | {
+      type: "wallet_switchStarknetChain"
+      params: SwitchStarknetChainParameter
+      result: boolean
+    }
+  | {
+      type: string
+      params: any
+      result: never
+    }
+
+interface IStarknetWindowObject {
+  id: string
+  name: string
+  version: string
+  icon: string
+  request: <T extends RpcMessage>(
+    call: Omit<T, "result">,
+  ) => Promise<T["result"]>
+  enable: (options?: { starknetVersion?: "v3" | "v4" }) => Promise<string[]>
```

Comment:
what about `showModal`? `connect` matches a dapp with a wallet instance but `enable` is about actually requesting a wallet to connect with the dapp, and maybe a dapp would like to attempt to do so with a non-preauthorized wallet - yet silently

Reply:
you can tell if a wallet is preauthorized by calling `isPreauthorized`. 
`enable` never had any arguments for Argent X, silentconnect should be fully managed by the wallet connect lib, like get-starknet

Reply:
cool, we will do the same